polar2grid/writers/hdf5.py

```diff
@@ -0,0 +1,293 @@
+            compression = None
+
+        add_geolocation = kwargs.pop("add_geolocation") if "add_geolocation" in kwargs else False
+
+        # will this be written to one or multiple files?
+        output_names = []
+
+        for dataset_id in dataset:
+            args = self._output_file_kwargs(dataset_id, dtype)
+            out_filename = filename or self.get_filename(**args)
+            output_names.append(out_filename)
+        one_file = all_equal(output_names)
+
+        filename = output_names[0]
+        if not all_equal(output_names):
+            LOG.warning("Writing to {}".format(filename))
+
+        hdf5_fh = self.open_hdf5_filehandle(filename, append=append)
+
+        datasets_by_area = self.iter_by_area(dataset)
+        for area, data_arrs in datasets_by_area:
+            # Initialize source/targets at start of each new AREA grouping.
+            dsets = []
+            targets = []
+            # open hdf5 file handle, check if group already exists.
+            parent_group = self.create_proj_group(filename, hdf5_fh, area)
+
+            if add_geolocation:
+                chunks = data_arrs[0].chunks
+                geo_sets, fnames = self.write_geolocation(
+                    hdf5_fh, filename, parent_group, area, dtype, append, compression, chunks
+                )
+                dsets.append(geo_sets)
+                targets.append(fnames)
+
+            for data_arr in data_arrs:
+                d_dtype = data_arr.dtype if dtype is None else dtype
+                hdf_subgroup = "{}/{}".format(parent_group, data_arr.attrs["name"])
+
+                file_var = FakeHDF5(filename, hdf_subgroup)
+                arr_dset = self.check_variable(hdf5_fh, hdf_subgroup, data_arr.data, d_dtype, append, compression)
+                dsets.append(data_arr.data)
+                targets.append(file_var)
+        return (dsets, targets)
+
+
+def add_writer_argument_groups(parser, group=None):
+    """Create writer argument groups."""
+    from argparse import SUPPRESS
+
+    if group is None:
+        group = parser.add_argument_group(title="hdf5 Writer")
+    group.add_argument("--output-filename", dest="filename", help="Custom file pattern to save dataset to")
+    group.add_argument(
+        "--dtype",
+        choices=NumpyDtypeList(NUMPY_DTYPE_STRS),
+        type=str_to_dtype,
+        help="Data type of the output file (8-bit unsigned " "integer by default - uint8)",
+    )
+    group.add_argument(
+        "--fill-value", dest="fill_value", type=int_or_float, help="Fill value for invalid data in this dataset."
+    )
+    group.add_argument("--compress", default="LZW", help="File compression algorithm (DEFLATE, LZW, NONE, etc)")
+    group.add_argument(
+        "--chunks",
+        dest="chunks",
+        type=tuple,
+        help="Chunked storage of dataset. " "Default is chosen appropriate to dataset by h5py.",
+    )
+    group.add_argument(
+        "--add-geolocation",
+        dest="add_geolocation",
```

Comment:
Probably my fault, but this `dest=` is unnecessary in this argument.

Reply:
Direct copy, I am sure there is split opinion regarding blame...at least 50% my fault.  Not a problem, this is fixable.